### ISSUE TITLE
Problem with quoted semicolons

### ISSUE THREAD

**Issue by snoe** [parinfer/parinfer.js#56]
I've been working with hiccup this week and I found a problem in insert mode that keeps biting me. I'm not sure if #49 is describing existing bugs or just trying to find a better way of handling this stuff, but I wanted to raise an issue for this because it's coming up so often in my codebase.

Here's the test-case

in

``` in
(for [col columns]
  "|
  [:div.td {:style "max-width: 500px;"}])
```

out

``` out
(for [col columns]
  "
  [:div.td {:style "max-width: 500px;"}])
```

The test error is 

```
expected----------------------------------
(for [col columns]
  "
  [:div.td {:style "max-width: 500px;"}])
actual------------------------------------
(for [col columns]
  "
  [:div.td {:style "max-width: 500px);"}])
expected: "(for [col columns]\n  \"\n  [:div.td {:style \"max-width: 500px;\"}])"
  actual: "(for [col columns]\n  \"\n  [:div.td {:style \"max-width: 500px);\"}])"
    diff: - "(for [col columns]\n  \"\n  [:div.td {:style \"max-width: 500px;\"}])"
          + "(for [col columns]\n  \"\n  [:div.td {:style \"max-width: 500px);\"}])"
```

**Comment by shaunlebron** [comment]
Thanks for reporting this.  This uncovers another case where Parinfer can corrupt strings.

The problem is that inserting a quote before `";"` gives `"";"`, which is still a valid clojure expression, and Parinfer will operate on anything valid.

I'm not yet sure if this problem is solvable.  But I have no doubt that the information fundamental to and necessary for solving it is related to quote-directionality.  If we can capture the intended direction that these quotes are pointed, we may be able to come up with some rules for handling these intermediate states created by inserting or deleting quotes.

**Comment by shaunlebron** [comment]
There are two things that I think can help this case without introducing much complexity:
- a trapdoor to disable parinfer once odd number of quotes are found
- auto-closing strings to prevent trapdoor encounters

details here: https://github.com/shaunlebron/parinfer/wiki/The-Problem-with-Quotes

**Comment by shaunlebron** [comment]
this is fixed in the quotefix branch.  see [test cases here](https://github.com/shaunlebron/parinfer/blob/556c2b72cc8c58af9e4303d9a1b515b4ae598a73/lib/test/parinfer/cases/indent-mode.md#unbalanced-quotes-in-comments)

I'm going to finish #63 before cutting a release

**Comment by shaunlebron** [comment]
@snoe: sorry, forgot to follow-up on this, but this has been fixed for the past few releases (at 1.4.0 now).  Here's the demo of your case in atom-parinfer:

<a href="http://gfycat.com/AdmiredPerkyEland">
<img src="https://zippy.gfycat.com/AdmiredPerkyEland.gif"></a>
